### PR TITLE
Internal: Editor V2 - Support dark mode in Site Settings [ED-9951]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

**Review by rotemee — DISMISSED** [review]

File: packages/packages/editor/src/components/theme-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	elementor: {
+		settings: {
+			editorPreferences: {
+				model: {
+					get: ( key: 'ui_theme' ) => ColorScheme,
+				}
+			}
+		}
+	}
+}
```
Comment (Nevoss) [inline packages/packages/editor/src/components/theme-provider.tsx:27]: Make it short

File: packages/packages/editor/src/components/theme-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+					setColorScheme( getV1ColorScheme() );
+				}
+			}
+		);
+
+		return () => {
+			unsubscribeFromV1Ready();
+			unsubscribeFromCommand();
+		};
+	}, [] );
```
Comment (Nevoss) [inline packages/packages/editor/src/components/theme-provider.tsx:63]: separate effects

File: packages/packages/editor/src/components/theme-provider.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+				const event = e as CommandEvent<{
+					container: {
+						id: string,
+					},
+					settings: {
+						ui_theme?: ColorScheme,
+					},
+				}>;
+
+				const isUserPreferences = event.args?.container?.id === 'editorPreferences_settings';
```
Comment (Nevoss) [inline packages/packages/editor/src/components/theme-provider.tsx:50]: remove

File: packages/packages/editor/src/components/theme-provider.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+				const isColorScheme = event.args?.settings && 'ui_theme' in event.args.settings;
```
Comment (Nevoss) [inline packages/packages/editor/src/components/theme-provider.tsx:51]: Explain

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: packages/packages/editor/src/components/theme-provider.tsx (hunk 2 above)
Comment (StyleShit) [reply packages/packages/editor/src/components/theme-provider.tsx:63]: Done

File: packages/packages/editor/src/components/theme-provider.tsx (hunk 3 above)
Comment (StyleShit) [reply packages/packages/editor/src/components/theme-provider.tsx:50]: Done

**Review by Nevoss — APPROVED** [review]

**Pull request merged**